sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -91,3 +91,8 @@ def __init__(self, subject, event_type, data, data_version, **kwargs):
         kwargs.setdefault("data_version", data_version)
 
         super(EventGridEvent, self).__init__(**kwargs)
+    
+    def __repr__(self):
+        return "EventGridEvent(subject={}, event_type={}, id={}, event_time={})".format(
+            self.subject, self.event_type, self.id, self.event_time
+        )[:1024]
```

Comment:
dumb question: only those four properties are needed for `__repr__`?
what about the other ivars like `data`, `data_version` etc.

Reply:
The purpose of __repr__  is to give a small gist of what the object is with important information - and trimmed to 1024 characters instead of some random hexcode. It's NOT to give the complete info.
data and data_version aren't the most important aspects of an event

Reply:
out of curiosity, a quick follow-up: why is `data` not considered an important aspect of an event?

Reply:
just to be clear - it's TYPICALLY not THE most important aspect in an event when representing the event object in the sense that having data doesn't provide as much value as providing subject and id given we only want 1024 characters in the repr

Reply:
consistency with CloudEvent repr makes sense!